lib/js_defer.js

```diff
@@ -1456,6 +1462,29 @@ deferJsNs.DeferJs.prototype.noDeferCreateElementOverride = function() {
   };
 };
 
+deferJsNs.DeferJs.prototype.deferInsertScriptTag = function() {
+  var me = this;
+  Element.prototype.insertBefore = function (newElement, refElement) {
+    if (me.overrideDefaultImplementation_ &&
+      newElement.nodeName === 'SCRIPT' &&
+      newElement.getAttribute('src')) {
+        me.log('Defer insertion of ' + newElement.getAttribute('src'));
+        me.originInsertBeforeQueue_.push([this, newElement, refElement]);
+    } else {
+      me.originInsertBefore_.apply(this, arguments);
+    }
+  };
+}
+
+deferJsNs.DeferJs.prototype.insertScriptInQueue = function() {
```

Comment:
This will be handled by `runNext`. If we use the `submitTask` `runNext` pattern we can avoid the dependency on the `onComplete` event.